fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -196,56 +198,10 @@ public View getView(String dbName, String viewName) {
     }
 
     @Override
-    public List<String> listPartitionNames(String dbName, String tableName, long snapshotId, ExecutorService executorService) {
-        IcebergTableName icebergTableName = new IcebergTableName(dbName, tableName, snapshotId);
-        if (partitionNames.asMap().containsKey(icebergTableName)) {
-            return partitionNames.getIfPresent(icebergTableName);
-        } else {
-            org.apache.iceberg.Table icebergTable = delegate.getTable(dbName, tableName);
-            List<String> partitionNames = Lists.newArrayList();
-
-            if (icebergTable.specs().values().stream().allMatch(PartitionSpec::isUnpartitioned)) {
-                return partitionNames;
-            }
-
-            if (snapshotId == -1) {
-                if (icebergTable.currentSnapshot() == null) {
-                    return partitionNames;
-                } else {
-                    snapshotId = icebergTable.currentSnapshot().snapshotId();
-                }
-            }
-
-            partitionNames = listPartitionNamesWithSnapshotId(icebergTable, dbName, tableName, snapshotId, executorService);
-            this.partitionNames.put(icebergTableName, partitionNames);
-            return partitionNames;
-        }
-    }
-
-    private List<String> listPartitionNamesWithSnapshotId(
-            Table table, String dbName, String tableName, long snapshotId, ExecutorService executorService) {
-        Set<String> partitionNames = Sets.newHashSet();
-        StarRocksIcebergTableScanContext scanContext = new StarRocksIcebergTableScanContext(
-                catalogName, dbName, tableName, PlanMode.LOCAL);
-        scanContext.setOnlyReadCache(true);
-        TableScan tableScan = getTableScan(table, scanContext)
-                .planWith(executorService)
-                .useSnapshot(snapshotId);
-
-        try (CloseableIterable<FileScanTask> fileScanTaskIterable = tableScan.planFiles();
-                CloseableIterator<FileScanTask> fileScanTaskIterator = fileScanTaskIterable.iterator()) {
-
-            while (fileScanTaskIterator.hasNext()) {
-                FileScanTask scanTask = fileScanTaskIterator.next();
-                StructLike partition = scanTask.file().partition();
-                partitionNames.add(convertIcebergPartitionToPartitionName(scanTask.spec(), partition));
-            }
-        } catch (IOException e) {
-            throw new StarRocksConnectorException(String.format("Failed to list iceberg partition names %s.%s",
-                    dbName, tableName), e);
-        }
-
-        return new ArrayList<>(partitionNames);
+    public Map<String, Partition> loadPartitions(String dbName, String tableName, long snapshotId,
+                                                 ExecutorService executorService) {
+        IcebergTableName key = new IcebergTableName(dbName, tableName, snapshotId);
+        return partitionCache.getUnchecked(key);
```

Comment:
return delegate.loadPartitions(key.dbName, key.tableName, key.snapshotId,executorService);

Reply:
Why? We have cache here?  partitionCache will do `delegate.loadPartitions`.